hail/src/main/scala/is/hail/expr/ir/TypeCheck.scala

```diff
@@ -152,6 +152,15 @@ object TypeCheck {
         assert(nd.typ.isInstanceOf[TNDArray])
         assert(nd.typ.asInstanceOf[TNDArray].nDims == idxs.length)
         assert(idxs.forall(_.typ.isOfType(TInt64())))
+      case x@NDArraySlice(nd, slices) =>
+        assert(nd.typ.isInstanceOf[TNDArray])
+        val childTyp =nd.typ.asInstanceOf[TNDArray]
+        assert(childTyp.nDims > 0)
```

Comment:
Is this necessary? `NDArraySlice(zeroDimArray, ())` seems consistent to me.

Reply:
True that is consistent